src/Compilers/Core/Portable/Binding/BindingDiagnosticBag.cs

```diff
@@ -152,7 +132,40 @@ internal void AddDependency(TAssemblySymbol? dependency)
 
         internal void AddDependencies(ICollection<TAssemblySymbol>? dependencies)
         {
-            if (dependencies?.Count > 0 && DependenciesBag is object)
+            if (!dependencies.IsNullOrEmpty() && DependenciesBag is object)
+            {
+                foreach (var candidate in dependencies)
+                {
+                    DependenciesBag.Add(candidate);
+                }
+            }
+        }
+
+        internal void AddDependencies(IReadOnlyCollection<TAssemblySymbol>? dependencies)
```

Comment:
Consider unifying this and the one above it with an `AddDependencies(IEnumerable<TAssemblySymbol>? dependencies)`

Reply:
Actually, I suppose you can't because you need to know how many elements are in it. It's a shame that ICollection doesn't implement IReadonlyCollection.

---
In reply to: [362668093](https://github.com/dotnet/roslyn/pull/40640#discussion_r362668093) [](ancestors = 362668093)